app/helpers/annotation_categories_helper.rb

```diff
@@ -3,16 +3,46 @@ module AnnotationCategoriesHelper
   def prepare_for_conversion(annotation_categories)
     result = {}
     annotation_categories.each do |annotation_category|
-      annotation_texts = []
-      annotation_category.annotation_texts.each do |annotation_text|
-        annotation_texts.push(annotation_text.content)
+      if annotation_category.flexible_criterion.nil?
+        annotation_texts = []
+        annotation_texts.push(nil)
+        annotation_category.annotation_texts.each do |annotation_text|
+          annotation_texts.push(annotation_text.content)
+        end
+        result[annotation_category.annotation_category_name] = annotation_texts
+      else
+        annotation_text_info = []
+        annotation_text_info.push(annotation_category.flexible_criterion.name)
+        annotation_category.annotation_texts.each do |annotation_text|
+          annotation_text_info.push(annotation_text.content)
+          annotation_text_info.push(annotation_text.deduction.to_s)
+        end
+        result[annotation_category.annotation_category_name] = annotation_text_info
       end
-      result[annotation_category.annotation_category_name] = annotation_texts
     end
     result
   end
 
   def convert_to_yml(annotation_categories)
-    prepare_for_conversion(annotation_categories).ya2yaml
+    categories_data = {}
+    annotation_categories.each do |category|
+      if category.flexible_criterion_id.nil?
+        annotation_texts = []
+        category.annotation_texts.each do |annotation_text|
+          annotation_texts.push(annotation_text.content)
+        end
+        categories_data[category.annotation_category_name] = annotation_texts
```

Comment:
In fact, you can probably do this all in one line (without using a temporary `annotation_texts` variable. Similarly below.